### DISCUSSION TITLE
jdk14?

### DISCUSSION THREAD

**Discussion by viqtor** [dadoonet/elasticsearch-beyonder#181] [Q&A] [answered]
whats the reasoning behind moving this to jdk14? #161 for reference.
according to [support matrix](https://www.elastic.co/support/matrix#matrix_jvm) elastic still supports jdk8 in the most recent version where jdk14 isn't officially listed as supported in 7.13

**Comment by dadoonet** [comment, marked as answer]
I just wanted to remove java8 from this project as IMO it should not be used anymore in production.

Is that an issue on your side?

**Reply by viqtor** [reply]
Not java8 no, we are running java11 but its the same issue. With this change it requires java14.
Would it be an alternative to compile with a target of 8? At least until elastic drops supports for it?

**Reply by dadoonet** [reply]
Yeah. This is doable. I just need to check what won't compile anymore with java8 and adjust.

would you like to work on a PR?

**Reply by viqtor** [reply]
Sure thing. Opened https://github.com/dadoonet/elasticsearch-beyonder/pull/182 you seemed to use JDK10 features in ResourceList so picked the lowest LTS version after that (11).

**Reply by dadoonet** [reply]
Wondering if we should switch back this code to Java8. WDYT?

**Reply by viqtor** [reply]
Sounds like a good plan at least until elastic drops support for it.
